Comments tell that LANG has to be the first value, and LC_ALL the last
one.  Thus LANGUAGE is printed between them.

# DP: Dpatch author: Denis Barbier
# DP: Patch author: Denis Barbier
# DP: Upstream status: not submitted
# DP: Date: 2006-01-08

Index: locale/programs/locale.c
===================================================================
--- locale/programs/locale.c.orig	2007-10-23 08:46:19.000000000 -0400
+++ locale/programs/locale.c	2007-10-23 08:47:08.000000000 -0400
@@ -752,6 +752,7 @@
 {
   size_t cat_no;
   const char *lcall = getenv ("LC_ALL");
+  const char *language = getenv ("LANGUAGE");
   const char *lang = getenv ("LANG") ? : "";
 
   auto void get_source (const char *name);
@@ -770,6 +771,9 @@
   /* LANG has to be the first value.  */
   printf ("LANG=%s\n", lang);
 
+  if (language != NULL && getenv ("POSIXLY_CORRECT") == NULL)
+    printf ("LANGUAGE=%s\n", language);
+
   /* Now all categories in an unspecified order.  */
   for (cat_no = 0; cat_no < NCATEGORIES; ++cat_no)
     if (cat_no != LC_ALL)
